observe/obervable.go

```diff
@@ -0,0 +1,164 @@
+package observe
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/m3db/m3x/close"
+)
+
+var ErrClosed = errors.New("closed")
+
+// Observer observes an Observable instance, can get notification when the Observable updates
+type Observer interface {
+	xclose.SimpleCloser
+
+	// C returns the notification channel
+	C() <-chan struct{}
+	// Get returns the latest value of the Observable instance
+	Get() interface{}
+}
+
+// Observable can be observed by Observers
+type Observable interface {
+	xclose.SimpleCloser
+
+	// Get returns the latest value
+	Get() interface{}
+	// Subscribe returns an Observer that will be notified on updates
+	Subscribe() (Observer, error)
```

Comment:
Might be better to do `GetAndSubscribe` instead to return the initial value along with the observer so the observer doesn't need to call `Get` to get it